drivers/input/input_mcux_kpp.c

```diff
@@ -0,0 +1,198 @@
+	const struct pinctrl_dev_config *pcfg;
+};
+
+struct kpp_data {
+	uint32_t clock_rate;
+	struct k_work_delayable work;
+	uint8_t read_keys_old[KPP_KEYPAD_COLUMNNUM_MAX];
+	uint8_t read_keys_new[KPP_KEYPAD_COLUMNNUM_MAX];
+	uint8_t key_pressed_number;
+	const struct device *dev;
+};
+
+static void get_source_clk_rate(const struct device *dev, uint32_t *clk_rate)
+{
+	const struct kpp_config *dev_cfg = dev->config;
+	const struct device *ccm_dev = dev_cfg->ccm_dev;
+	clock_control_subsys_t clk_sub_sys = dev_cfg->clk_sub_sys;
+
+	if (!device_is_ready(ccm_dev)) {
+		LOG_ERR("CCM driver is not installed");
+		*clk_rate = 0;
+		return;
+	}
+
+	clock_control_get_rate(ccm_dev, clk_sub_sys, clk_rate);
+}
+
+static void kpp_work_handler(struct k_work *work)
+{
+	struct k_work_delayable *dwork = k_work_delayable_from_work(work);
+	struct kpp_data *drv_data = CONTAINER_OF(dwork, struct kpp_data, work);
+	const struct device *dev = drv_data->dev;
+	const struct kpp_config *config = dev->config;
+
+	uint8_t read_keys_new[KPP_KEYPAD_COLUMNNUM_MAX];
+
+	/* Read the key press data */
+	KPP_keyPressScanning(config->base, read_keys_new, drv_data->clock_rate);
+
+	/* Analyze the keypad data */
+	for (int col = 0; col < INPUT_KPP_COLUMNNUM_MAX; col++) {
+		if (drv_data->read_keys_old[col] == read_keys_new[col]) {
+			continue;
+		} else {
+			for (int row = 0; row < INPUT_KPP_ROWNUM_MAX; row++) {
+				if (((drv_data->read_keys_old[col] ^ read_keys_new[col])
+					& BIT(row)) == 0) {
+					continue;
+				} else {
+					if ((read_keys_new[col] & BIT(row)) != 0) {
+						/* Key press event */
+						KPP_SetSynchronizeChain(config->base,
+							kKPP_ClearKeyDepressSyncChain);
+						input_report_abs(dev, INPUT_ABS_X, col,
+							false, K_FOREVER);
+						input_report_abs(dev, INPUT_ABS_Y, row,
+							false, K_FOREVER);
+						input_report_key(dev, INPUT_BTN_TOUCH, 1,
+							true, K_FOREVER);
+						drv_data->key_pressed_number++;
+					} else {
+						/* Key release event */
+						KPP_SetSynchronizeChain(config->base,
+							kKPP_SetKeyReleasesSyncChain);
+						input_report_abs(dev, INPUT_ABS_X, col,
+							false, K_FOREVER);
+						input_report_abs(dev, INPUT_ABS_Y, row,
+							false, K_FOREVER);
+						input_report_key(dev, INPUT_BTN_TOUCH, 0,
+							true, K_FOREVER);
+						drv_data->key_pressed_number--;
+					}
+				}
+			}
+			drv_data->read_keys_old[col] = read_keys_new[col];
+		}
+	}
+
+	if (drv_data->key_pressed_number == 0U) {
+		KPP_ClearStatusFlag(config->base, kKPP_keyDepressInterrupt |
+			kKPP_keyReleaseInterrupt);
+		KPP_EnableInterrupts(config->base, kKPP_keyDepressInterrupt);
+	} else {
+		k_work_schedule(&drv_data->work, K_MSEC(CONFIG_INPUT_KPP_PERIOD_MS));
+	}
+}
+
+static void kpp_isr(const struct device *dev)
+{
+	const struct kpp_config *config = dev->config;
+	struct kpp_data *drv_data = dev->data;
+
+	uint16_t status = KPP_GetStatusFlag(config->base);
+
+	if ((status & kKPP_keyDepressInterrupt) != 0) {
```

Comment:
flip the condition, usual pattern, normal exdcution code inline, error and other special cases in branches

```c
	if ((status & kKPP_keyDepressInterrupt) == 0) {
		LOG_ERR("No key press or release detected");
                return;
	}

        drv_data->key_pressed_number = 0;
        ...